tasks.py

```diff
@@ -37,3 +44,20 @@ def collect_over_thirty(self) -> List[Dict]:
                 self.over_thirty_records.append(record)
 
         return self.over_thirty_records
+
+
+class CompanyFieldGenerator:
+    def __init__(self, user_records, company_records) -> None:
+        self.user_records = user_records
+        self.company_records = company_records
+
+    def company_id_to_field(self) -> List[Dict]:
+        for record in self.user_records:
+            if "company_id" in record:
+                matching_company = list(filter(lambda match: match["id"] == record["company_id"], self.company_records))
```

Comment:
Better. It could be improved by creating a company_by_id dictionary before the loop and use that in the comparison:
```
company_by_id = {company['id']: company for company in self.company_records.items() if company['id']}
...
matching_company = company_by_id.get(record['company_id'])
```

Reply:
oh! That makes a lot of sense, I'll keep this method in mind, definitely!